### PR TITLE
Add this parameter types to Mocha callbacks.

### PR DESCRIPTION
**Pull request by mgroenhoff**

### CODE REVIEW

File: mocha/index.d.ts [hunk 1]
```diff
@@ -110,6 +106,24 @@ declare class Mocha {
 
 // merge the Mocha class declaration with a module
 declare namespace Mocha {
+    interface ISuiteCallbackContext {
```
Comment (anubmat) [inline mocha/index.d.ts]: Don't begin interface names with 'I'.

Comment (mgroenhoff) [inline mocha/index.d.ts]: I just used the convention used in the entire file. I can change it all let existing applications break that rely on the interface type names.

**Comment by vvakame** [comment]
*bardjs/index.d.ts*

to author (@TepigMC). Could you review this PR?
:+1: or :-1:?

Checklist

* [ ] pass the Travis CI test?

------

*mocha/index.d.ts*

to authors (@kazimanzurrashid @otiai10 @jt000 @enlight). Could you review this PR?
:+1: or :-1:?

Checklist

* [ ] pass the Travis CI test?

**Comment by TepigMC** [comment]
I am not super familiar with Mocha, I just made the type definition file based off their source and docs to scratch an itch. I don't use Mocha anymore and could not test this, so, if it works, you have my approval.

**Comment by vvakame** [comment]
We can't get response from other author. please ping me after 3 days. I'll merge this PR.

File: mocha/index.d.ts (hunk 1 above)
Comment (jt000) [inline mocha/index.d.ts]: :+1: for sticking w/ the convention of the file & starting these interfaces with an I

**Review by jt000 — APPROVED** [review]

**Comment by jt000** [comment]
👍 LGTM - Sorry for the delay

**Comment by mgroenhoff** [comment]
@vvakame It's been a little over 3 days ;)

**Comment by vvakame** [comment]
wow, I'm so sorry. I'm moving my home to new place... It merged. thank you!

**Comment by mgroenhoff** [comment]
No worries. Thank you!

**Pull request merged**